utils/file.js

```diff
@@ -76,14 +78,16 @@ exports.fileSystem = (function(){
 
       while (null !== (chunk = readable.read(chunkSize))) {
         const chunkId = sha1HashData(chunk);
-        manifest.chunks.push(chunkId);
+        manifest.chunks[chunkId] = [];
 
+        copyShards(chunk, chunkId, manifest)
         storeShards(chunk, chunkId)
```

Comment:
It doesn't look like we're creating (writing to disk) the redundant shard files - just the original shard. That could be by design, though. Is that what we want here? @floalex @WilfredTA

Reply:
Ah, that make sense - I do remember this being the strategy. Not sure if I'm missing something or it's already been discussed, but is it seems life if the node being referenced (the one that actually holds the file path and file) went down, then the three other nodes still being up isn't as useful. I don't have any alternatives, but it just made me think. Maybe something to discuss, but everything in the code here is fine.

Reply:
@dylankb Three different nodes will still store actual copies of the shard. Each copy of the shard stored on another node will have the *exact same* content but a unique ID, that's all. If one node goes down, there are still two other nodes storing other copies of the shard